### ISSUE TITLE
Google Pay button doesn't work in Compose Preview

### ISSUE THREAD

**Issue by barry-irvine** [google-pay/compose-pay-button#14]
Whenever I add the GPay button to a Compose Preview, the Preview crashes and doesn't get rendered. The stacktrace reads as below. I have added the google play services version to the Manifest as it requests. To work around this issue I've added the following lines to my GPay Button Composable

```
@Composable
fun GPayButton() {
    // Early return if in preview
    if (LocalInspectionMode.current) {
        Button(onClick = {}, Modifier.fillMaxWidth()) {
            Text("Google Pay Button")
        }
        return
    }
    PayButton(...)
}
```

```
2023-12-18 10:32:37,480 [235377129]   WARN - #com.android.tools.idea.uibuilder.scene.LayoutlibSceneManager - A required meta-data tag in your app's AndroidManifest.xml does not exist.  You must have the following declaration within the <application> element:     <meta-data android:name="com.google.android.gms.version" android:value="@integer/google_play_services_version" />
com.google.android.gms.common.GooglePlayServicesMissingManifestValueException: A required meta-data tag in your app's AndroidManifest.xml does not exist.  You must have the following declaration within the <application> element:     <meta-data android:name="com.google.android.gms.version" android:value="@integer/google_play_services_version" />
	at com.google.android.gms.common.GooglePlayServicesUtilLight.isGooglePlayServicesAvailable(com.google.android.gms:play-services-basement@@18.1.0:8)
	at com.google.android.gms.common.GoogleApiAvailabilityLight.isGooglePlayServicesAvailable(com.google.android.gms:play-services-basement@@18.1.0:2)
	at com.google.android.gms.common.GoogleApiAvailability.isGooglePlayServicesAvailable(com.google.android.gms:play-services-base@@18.2.0:2)
	at com.google.android.gms.wallet.button.PayButton.initialize(com.google.android.gms:play-services-wallet@@19.2.1:11)
	at com.google.pay.button.PayButtonKt$PayButton$1$1.invoke(PayButton.kt:66)
	at com.google.pay.button.PayButtonKt$PayButton$1$1.invoke(PayButton.kt:64)
	at androidx.compose.ui.viewinterop.ViewFactoryHolder.<init>(AndroidView.android.kt:337)
	at androidx.compose.ui.viewinterop.AndroidView_androidKt$createAndroidViewNodeFactory$1.invoke(AndroidView.android.kt:271)
	at androidx.compose.ui.viewinterop.AndroidView_androidKt$createAndroidViewNodeFactory$1.invoke(AndroidView.android.kt:270)
	at androidx.compose.ui.viewinterop.AndroidView_androidKt$AndroidView$$inlined$ComposeNode$1.invoke(Composables.kt:254)
	at androidx.compose.runtime.ComposerImpl$createNode$2.invoke(Composer.kt:1610)
	at androidx.compose.runtime.ComposerImpl$createNode$2.invoke(Composer.kt:1608)
	at androidx.compose.runtime.ComposerImpl$recordInsert$2.invoke(Composer.kt:3546)
	at androidx.compose.runtime.ComposerImpl$recordInsert$2.invoke(Composer.kt:3543)
	at androidx.compose.runtime.CompositionImpl.applyChangesInLocked(Composition.kt:818)
	at androidx.compose.runtime.CompositionImpl.applyChanges(Composition.kt:849)
	at androidx.compose.runtime.Recomposer.composeInitial$runtime_release(Recomposer.kt:1041)
	at androidx.compose.runtime.CompositionImpl.setContent(Composition.kt:520)
	at androidx.compose.ui.platform.WrappedComposition$setContent$1.invoke(Wrapper.android.kt:142)
	at androidx.compose.ui.platform.WrappedComposition$setContent$1.invoke(Wrapper.android.kt:133)
	at androidx.compose.ui.platform.AndroidComposeView.setOnViewTreeOwnersAvailable(AndroidComposeView.android.kt:1191)
	at androidx.compose.ui.platform.WrappedComposition.setContent(Wrapper.android.kt:133)
	at androidx.compose.ui.platform.WrappedComposition.onStateChanged(Wrapper.android.kt:183)
	at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.kt:314)
	at androidx.lifecycle.LifecycleRegistry.addObserver(LifecycleRegistry.kt:192)
	at androidx.compose.ui.platform.WrappedComposition$setContent$1.invoke(Wrapper.android.kt:140)
	at androidx.compose.ui.platform.WrappedComposition$setContent$1.invoke(Wrapper.android.kt:133)
	at androidx.compose.ui.platform.AndroidComposeView.onAttachedToWindow(AndroidComposeView.android.kt:1266)
	at android.view.View.dispatchAttachedToWindow(View.java:21291)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3491)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3498)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3498)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3498)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3498)
	at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3498)
	at android.view.AttachInfo_Accessor.setAttachInfo(AttachInfo_Accessor.java:54)
	at com.android.layoutlib.bridge.impl.RenderSessionImpl.inflate(RenderSessionImpl.java:372)
	at com.android.layoutlib.bridge.Bridge.createSession(Bridge.java:450)
	at com.android.tools.idea.layoutlib.LayoutLibrary.createSession(LayoutLibrary.java:122)
	at com.android.tools.rendering.RenderTask.createRenderSession(RenderTask.java:742)
	at com.android.tools.rendering.RenderTask.lambda$inflate$7(RenderTask.java:889)
	at com.android.tools.rendering.RenderExecutor$runAsyncActionWithTimeout$3.run(RenderExecutor.kt:202)
	at com.android.tools.rendering.RenderExecutor$PriorityRunnable.run(RenderExecutor.kt:316)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```

**Comment by dmengelt** [comment]
Hi @barry-irvine thanks for reporting this.

We just released `19.3.0-beta01` of the main `play-services-wallet` SDK. This added basic support for previewing the button. We plan to release `19.3.0` in early 2024 and afterwards we will update the compose pay button to use the new version of the dependency.

If you cannot wait feel free to try the following:
```
implementation ("com.google.pay.button:compose-pay-button:0.1.3") {
    exclude group: 'com.google.android.gms', module: 'play-services-wallet'
}
implementation 'com.google.android.gms:play-services-wallet:19.3.0-beta01'
```

**Comment by dmengelt** [comment]
@barry-irvine version `0.1.4` added support for basic `@Preview` functionality:

<img width="1158" alt="Screenshot 2024-02-27 at 12 08 28" src="https://github.com/google-pay/compose-pay-button/assets/4389374/86cd4210-9b93-44cd-a1f7-26ccf6dc1ce9">
